### PR TITLE
add publishTimestamp to Source

### PR DESCRIPTION
**Pull request by shyba**

### CODE REVIEW

**Comment by shyba** [comment]
@eukreign @jackrobison with this last commit it passes the [regression tests](https://github.com/lbryio/lbryschema/pull/41#issuecomment-386739690). Maybe there is a better way to do this.

File: lbryschema/encoding.py [hunk 1]
```diff
@@ -76,3 +76,14 @@ def decode_b64_fields(claim_dictionary):
         claim_dictionary[SIGNATURE]['certificateId'] = encoded_cert_id
     claim_dictionary[claim_type] = claim_value
     return claim_dictionary
+
+
+def filter_optional_fields_from_dictionary(claim_dictionary):
```
Comment (jackrobison) [inline lbryschema/encoding.py]: I think this should done by bumping the version of the proto being changed - then you also don't need to clear the field.

File: lbryschema/proto/source.proto [hunk 2]
```diff
@@ -13,4 +13,6 @@ message Source {
     required SourceTypes sourceType = 2;
     required bytes source = 3;
     required string contentType = 4;
+
+    optional sint64 publishTime = 5;
```
Comment (jackrobison) [inline lbryschema/proto/source.proto]: I'm of the opinion this should be in `Metadata` (but I'd put `size` in `Source`). However there's an argument to be made that `Source` is the correct place. In any case, we need a clear distinction of which kinds of properties go where.

**Comment by tzarebczan** [comment]
Relies on claim signature discussion: https://github.com/lbryio/internal-issues/issues/182

**Comment by kauffj** [comment]
@eukreign was the final call here to add both publishTime and releaseTime? Just making sure we got the naming on this one right.

**Comment by eukreign** [comment]
We have bumped this PR and the related PR in lbrynet to come after 0.21.0 so that @shyba and I can work out the best way forward (specifically we need to work out how to do the claim signature changes). When we come back to this issue we'll figure out the best naming. In the meantime any feedback and comments would be greatly appreciated.

**Comment by eukreign** [comment]
This repo is now merged into lbry repo.

**Pull request closed without merging**